core/controlplane/config/templates/cloud-config-controller

```diff
@@ -991,7 +991,7 @@ write_files:
           -ec \
           'echo decrypting assets
            shopt -s nullglob
-           for encKey in /etc/kubernetes/{ssl,{{if .AssetsConfig.HasAnyAuthTokens}}auth{{end}}}/*.enc; do
+           for encKey in /etc/kubernetes/{ssl,{{ if or (.AssetsConfig.HasAuthTokens) ( and .Experimental.TLSBootstrap.Enabled .AssetsConfig.HasTLSBootstrapToken) }}auth{{end}}}/*.enc; do
```

Comment:
@jollinshead I think this logic can be replaced as well without breaking the intended behavior.